### ISSUE TITLE
Random fail with account locking

### ISSUE THREAD

**Issue by jbfarez** [segmentio/aws-okta#125]
## Overview

Hello there,

I'm coming here today because, randomly and approximately 1 time over 10, when I run a command through `aws-okta` I receive an authentification error:

```
Failed to authenticate with okta: &errors.errorString{s:"POST https://my-company.okta.com/api/v1/authn: 401 Unauthorized"}
```

## How to reproduce

* Run the following command for about 10 times
```
 aws-okta exec my-profile -- aws s3 ls s3://
```

## Setup

 - `OS`: Ubuntu 18.04.1 LTS
 - `Release`:    18.04
 - `aws-okta` version: dev

## Details

In consequence, my Okta account is regularly blocked and you must unblock it to me or I have to wait to be unlocked.

## Additional informations

 -  `~/.aws/config` content:

```
[default]
region = eu-west-1
output = json

[profile okta]
aws_saml_url = home/amazon_aws/1234567890

[profile main]
region = eu-west-1
role_arn = arn:aws:iam::1234567890:role/my-role
```

 - AWS related environment variables:

```
AWS_PROFILE=default
AWS_VAULT_BACKEND=kwallet
```

**Comment by nickatsegment** [comment]
I can't repro this myself. I personally run `aws-okta` commands all day long, sometimes in like 2s watch loops, and have never encountered this. Maybe you could do an HTTP capture to debug further. But this is gonna be a tough one

**Comment by jbfarez** [comment]
Hi Nick, thanks for your help, actually, to be honest, we are not able to reproduce it on different hosts neither 😞.

It seems to be due to the local environment, I'll have a look closer and keep you posted here.

**Comment by nickatsegment** [comment]
Huh, strange.

I'm going to close for now. Ping me if you want it reopened for more investigation.

**Comment by cmsmith7** [comment]
This happened to me and it happened when my corporate SSO Active Directory password expired which is sync'ed with Okta, after changing the password I could no longer use aws-okta commands and received the error "Failed to authenticate with okta: &errors.errorString{s:"POST https://newsuk.okta-emea.com/api/v1/authn: 401 Unauthorized"}" .. I rectified this by going through the "aws-okta add" process.  aws-okta must cache the okta password somewhere, which gets overwritten following the "add" process.

**Comment by kharthigeyan** [comment]
I faced the same error got fixed by running the below command:

`aws-okta add --domain <companyname>.okta.com`

Type the username and password in the prompt.

**Comment by cmsmith7** [comment]
Just to add .. I found another issue a while back on MAC's that the aws-okta add process does not have the ability in some cases to overwrite the Okta entries stored in your Keychain (Okta credentials) and this requires them to be manually removed.  "aws-okta add" process then re-creates them.
